packages/snjs/lib/services/features_service.ts

```diff
@@ -73,9 +77,20 @@ export class SNFeaturesService extends PureService<void> {
       ContentType.ExtensionRepo,
       async (changed, inserted) => {
         const items = [...changed, ...inserted];
-        await this.updateExtensionKeySetting(items);
+        if (this.sessionManager.getUser()) {
+          await this.updateExtensionKeySetting(items);
+        }
       }
     );
+
+    this.removeFullSyncCompletedObserver = this.syncService.addEventObserver(
+      async (eventName) => {
+        if (eventName === SyncEvent.FullSyncCompleted && this.sessionManager.getUser()) {
+          const extensionRepoItems = this.itemManager.getItems(ContentType.ExtensionRepo);
+          await this.updateExtensionKeySetting(extensionRepoItems);
+        }
+      }
+    )
   }
 
   public async updateExtensionKeySetting(extensionRepoItems: SNItem[]): Promise<void> {
```

Comment:
@mobitar not sure if the logic is okay here, I'm iterating over the extensionRepoItems but there's only one extension key user setting. Safe to assume there will only be one extension repo item? Should I filter them somehow?

Reply:
Hmm, users can have custom repos. In this case we only want to migrate our official repo. But I suppose it's safe to assume only we use that regex format in our URL, so we can keep as is.